lib/checks/navigation/heading-order-evaluate.js

```diff
@@ -4,17 +4,18 @@ import { isVisible } from '../../commons/dom';
 
 function getLevel(vNode) {
   const role = vNode.attr('role');
-  if (role && role.includes('heading')) {
-    const ariaHeadingLevel = vNode.attr('aria-level');
-    const level = parseInt(ariaHeadingLevel, 10);
-
-    // default aria-level for a heading is 2 if it is
-    // not set or set to an incorrect value
-    // @see https://www.w3.org/TR/wai-aria-1.1/#heading
-    if (isNaN(level) || level < 1 || level > 6) {
-      return 2;
-    }
+  const headingRole = role && role.includes('heading');
+  const ariaHeadingLevel = vNode.attr('aria-level');
+  const level = parseInt(ariaHeadingLevel, 10);
+
+  // default aria-level for a heading is 2 if it is
+  // not set or set to an incorrect value
+  // @see https://www.w3.org/TR/wai-aria-1.1/#heading
+  if ((headingRole && isNaN(level)) || level < 1 || level > 6) {
+    return 2;
+  }
 
+  if (level) {
```

Comment:
This will return the `aria-level` for non-header elements like `<iframe aria-level="2"></iframe>`, when we actually need the iframe to be level=-1

Reply:
The rule doesn't run on iframes, as far as I can tell.

Reply:
The rule doesn't, but we use a selector inside the evaluate to look for iframes and assign them a level of -1 so in the after we can reconstruct heading order correctly for all iframes on the page. https://github.com/dequelabs/axe-core/blob/0d6604b74118c11546e376987323863ef1751e78/lib/checks/navigation/heading-order-evaluate.js#L40-L46